temporalio/plugin.py

```diff
@@ -0,0 +1,256 @@
+"""Plugin module for Temporal SDK.
+
+This module provides plugin functionality that allows customization of both client
+and worker behavior in the Temporal SDK through configurable parameters.
+"""
+
+from contextlib import AbstractAsyncContextManager, asynccontextmanager
+from typing import (
+    Any,
+    AsyncIterator,
+    Awaitable,
+    Callable,
+    Optional,
+    Sequence,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import temporalio.client
+import temporalio.converter
+import temporalio.worker
+from temporalio.client import ClientConfig, WorkflowHistory
+from temporalio.service import ConnectConfig, ServiceClient
+from temporalio.worker import (
+    Replayer,
+    ReplayerConfig,
+    Worker,
+    WorkerConfig,
+    WorkflowReplayResult,
+    WorkflowRunner,
+)
+
+T = TypeVar("T")
+
+PluginParameter = Union[None, T, Callable[[Optional[T]], T]]
+
+
+class SimplePlugin(temporalio.client.Plugin, temporalio.worker.Plugin):
+    """A simple plugin definition which limits has a limited set of configurations but makes it easier to produce
+    a simple plugin which needs to configure them.
+    """
```

Comment:
I think removing a couple words provides a little clarity in this docstring.
```suggestion
    """A simple plugin definition which has a limited set of configurations but makes it easier to produce
    a plugin which needs to configure them.
    """
```

Reply:
Oh yeah, definitely meant to get rid of the word limits when I changed that.